framework/set/setRKE2K3sConfig.go

```diff
@@ -124,20 +89,54 @@ func SetRKE2K3s(clusterName, k8sVersion, psact string, nodePools []config.Nodepo
 		machineConfigBlockBody := machineConfigBlock.Body()
 
 		kind := hclwrite.Tokens{
-			{Type: hclsyntax.TokenIdent, Bytes: []byte(`rancher2_machine_config_v2.rancher2_machine_config_v2.kind`)},
+			{Type: hclsyntax.TokenIdent, Bytes: []byte("rancher2_machine_config_v2.rancher2_machine_config_v2.kind")},
 		}
 
 		machineConfigBlockBody.SetAttributeRaw("kind", kind)
 
 		name := hclwrite.Tokens{
-			{Type: hclsyntax.TokenIdent, Bytes: []byte(`rancher2_machine_config_v2.rancher2_machine_config_v2.name`)},
+			{Type: hclsyntax.TokenIdent, Bytes: []byte("rancher2_machine_config_v2.rancher2_machine_config_v2.name")},
 		}
 
 		machineConfigBlockBody.SetAttributeRaw("name", name)
 
 		count++
 	}
 
+	upgradeStrategyBlock := rkeConfigBlockBody.AppendNewBlock("upgrade_strategy", nil)
+	upgradeStrategyBlockBody := upgradeStrategyBlock.Body()
+
+	upgradeStrategyBlockBody.SetAttributeValue("control_plane_concurrency", cty.StringVal(("10%")))
+	upgradeStrategyBlockBody.SetAttributeValue("worker_concurrency", cty.StringVal(("10%")))
+
+	snapshotBlock := rkeConfigBlockBody.AppendNewBlock("etcd", nil)
+	snapshotBlockBody := snapshotBlock.Body()
+
+	snapshotBlockBody.SetAttributeValue("disable_snapshots", cty.BoolVal(terraformConfig.ETCD.DisableSnapshots))
+	snapshotBlockBody.SetAttributeValue("snapshot_schedule_cron", cty.StringVal(terraformConfig.ETCD.SnapshotScheduleCron))
+	snapshotBlockBody.SetAttributeValue("snapshot_retention", cty.NumberIntVal(int64(terraformConfig.ETCD.SnapshotRetention)))
+
+	if (terraformConfig.Module == ec2RKE2 || terraformConfig.Module == ec2K3s) && terraformConfig.ETCD.S3 != nil {
```

Comment:
maybe because omitempty is enabled, but does this not give an error if terraformConfig.ETCD is nil? might be fine as is since ETCD is a pointer, but just checking if that was tested.

Reply:
Yeah I tested that. It doesn't give an error, in fact, I had to have it this way because if you try to have S3 but not the additional check for it, that is what gives an error.